tests/core/training/converters/test_story_markdown_to_yaml_converter.py

```diff
@@ -65,3 +74,47 @@ async def test_stories_are_converted(tmpdir: Path):
             "      - value1\n"
             "      - value2\n"
         )
+
+
+async def test_test_stories(tmpdir: Path):
+    converted_data_folder = tmpdir / "converted_data"
+    os.mkdir(converted_data_folder)
+
+    test_data_folder = tmpdir / "tests"
+    os.makedirs(test_data_folder, exist_ok=True)
+    test_data_file = Path(test_data_folder / "test_stories.md")
+
+    simple_story_md = """
+    ## ask product
+    * faq: what is [Rasa X](product)?
+        - slot{"product": "x"}
+        - respond_faq
+        - action_set_faq_slot
+    """
+
+    with open(test_data_file, "w") as f:
+        f.write(simple_story_md)
+
+    await StoryMarkdownToYamlConverter().convert_and_write(
+        test_data_file, converted_data_folder
+    )
+
+    assert len(os.listdir(converted_data_folder)) == 1
+
+    with open(f"{converted_data_folder}/test_stories_converted.yml", "r") as f:
+        content = f.read()
+        assert content == (
+            f'version: "{LATEST_TRAINING_DATA_FORMAT_VERSION}"\n'
+            "stories:\n"
+            "- story: ask product\n"
+            "  steps:\n"
+            "  - intent: faq\n"
+            "    user: |\n"
```

Comment:
I checked and it's correct to expect the `|-` here, since `|` means there's a newline after the value.